internal/impl/amqp09/integration_test.go

```diff
@@ -80,11 +126,29 @@ input:
 		integration.StreamTestStreamParallelLossy(1000),
 		integration.StreamTestStreamParallelLossyThroughReconnect(1000),
 	)
-	suite.Run(
-		t, template,
-		integration.StreamTestOptSleepAfterInput(500*time.Millisecond),
-		integration.StreamTestOptSleepAfterOutput(500*time.Millisecond),
+
+	streamTestOptFuncs := []integration.StreamTestOptFunc{
+		integration.StreamTestOptSleepAfterInput(500 * time.Millisecond),
+		integration.StreamTestOptSleepAfterOutput(500 * time.Millisecond),
 		integration.StreamTestOptPort(resource.GetPort("5672/tcp")),
 		integration.StreamTestOptVarOne("false"),
+	}
+
+	suite.Run(
+		t,
+		getTemplate(),
+		streamTestOptFuncs...,
 	)
+
+	t.Cleanup(func() {
+		assertQueueStateFromRabbitMQManagementAPI(resource)
+	})
+}
+
+func TestIntegrationAMQP09WithoutQueueDeclareAutoDelete(t *testing.T) {
+	doSetupAndAssertions(false, t)
+}
+
+func TestIntegrationAMQP09WithQueueDeclareAutoDelete(t *testing.T) {
```

Comment:
Unfortunately this test is failing (hitting the timeout) and I can't work out why.

We end up at https://github.com/benthosdev/benthos/blob/3c94c306b5cd16562f642fd26a5a4887d24c04a9/internal/integration/stream_test_helpers.go#L598

Reply:
I'm running this fork for the purpose mentioned in the issue (siphon off a cloned stream of prod stats for testing) and I am intermittently seeing on one of the instances (always the most swamped one, might be unrelated) that Benthos eventually (silently) stops doing as required.

Config looks like this:

```json
http:
  address: 0.0.0.0:4196

input:
  label: "api_stat_import_stream_source"
  amqp_0_9:
    urls:
      - "amqp://${STAT_IMPORT_CLONE_WMT_MQ_USERNAME}:${STAT_IMPORT_CLONE_WMT_MQ_PASSWORD}@${STAT_IMPORT_CLONE_WMT_MQ_IP}:5672/"
    queue_declare:
      enabled: true
      durable: false
      auto_delete: true
    bindings_declare:
      - exchange: "APIStatImportServer_main_fanout"
    auto_ack: true

output:
  label: "api_stat_import_stream_destination"
  amqp_0_9:
    urls:
      - "amqp://${WMT_MQ_USERNAME}:${WMT_MQ_PASSWORD}@${WMT_MQ_IP}:5672/"
    exchange: "APIStatImportServer_main_fanout"
    exchange_declare:
      enabled: true
      type: fanout
      durable: true
```

If I look at the management UI for the RabbitMQ servers, the source (prod) still has a happy consumer and the queue isn't growing, so some part of Benthos is still draining the queue as expected- just not getting anything in the destination RabbitMQ instance (also seems to have happy connections).

No errors in the Benthos CLI, but I haven't yet tried cranking up logging.

Reply:
That test closes an existing consumer and starts a new one up, so you're probably seeing the result of the old consumer being torn down along with all the data, then the consumer starts up and the queue gets remade. I would just disable that test when `auto_delete: true` as the two are incompatible anyway and I wouldn't expect it to work.